src/main/java/net/mcreator/blockly/java/blocks/ProcedureCallBlock.java

```diff
@@ -81,30 +84,41 @@ public class ProcedureCallBlock implements IBlockGenerator {
 			if (master.getTemplateGenerator() != null) {
 				Map<String, Object> dataModel = new HashMap<>();
 				dataModel.put("procedure", procedure.getName());
+				dataModel.put("dependencies", procedure.getDependencies(master.getWorkspace()));
+
 				if (call_at) {
 					dataModel.put("x", xcode);
 					dataModel.put("y", ycode);
 					dataModel.put("z", zcode);
-					dataModel.put("dependencies", procedure.getDependencies(master.getWorkspace()));
-					String code = master.getTemplateGenerator()
-							.generateFromTemplate("_call_procedure_at.java.ftl", dataModel);
-					master.append(code);
+				}
+
+				if (master instanceof BlocklyToJava blocklyToJava
+						&& blocklyToJava.getEditorType() == BlocklyEditorType.COMMAND_ARG) {

```

Comment:
In case of command arg, you should also check for dependencies to make sure procedure does not request dependences that do not exist

Reply:
How do I check if the dependencies are correct? All dependencies of the procedure are added to the `BlocklyToCode`, so I can not check the differences between both.

Reply:
Check my commit fixing this, do any changes if you see they are needed :)

Reply:
I just added the procedure's name causing the problem, so it will be easier to find the source of the problem.